drivers/ps2/ps2_npcx_controller.c

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright (c) 2021 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_ps2_ctrl
+
+/**
+ * @file
+ * @brief Nuvoton NPCX PS/2 module (controller) driver
+ *
+ * This file contains the driver of PS/2 module (controller) which provides a
+ * hardware accelerator mechanism to handle both incoming and outgoing data.
+ * The hardware accelerator mechanism is shared by four PS/2 channels.
+ */
+
+#include <drivers/clock_control.h>
+#include <drivers/ps2.h>
+#include <dt-bindings/clock/npcx_clock.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ps2_npcx_ctrl, CONFIG_PS2_LOG_LEVEL);
+
+#define NPCX_PS2_CH_COUNT 4
+
+/*
+ * Set WDAT3-0 and clear CLK3-0 in the PSOSIG register to
+ * reset the shift mechanism.
+ */
+#define NPCX_PS2_SHIFT_MECH_RESET (uint8_t) ~NPCX_PSOSIG_CLK_MASK_ALL
```

Comment:
Do you need the cast? If so, can you drop the space after it?

Reply:
There is a compile warning if no cast. Drop the space.